### PR TITLE
(CDAP-1917) Added classes for representing plugin version.

### PR DESCRIPTION
**Pull request by chtyim**
This is the first PR for CDAP-1917 by adding a new class in API to represent version information for template plugin.

### CODE REVIEW

File: cdap-api/src/main/java/co/cask/cdap/api/templates/plugins/PluginVersion.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  }
+
+  /**
+   * Constructs an instance by parsing the given string.
+   *
+   * @param str the version stirng.
+   * @param matchSuffix if {@code true}, try to match the version pattern by the suffix of the string. Otherwise match
+   *                    the whole string.
+   */
+  public PluginVersion(String str, boolean matchSuffix) {
```
Comment (gokulavasan) [inline cdap-api/src/main/java/co/cask/cdap/api/templates/plugins/PluginVersion.java]: Didn't find matchSuffix to be true in any of the tests. Is that intended?

Comment (chtyim) [inline cdap-api/src/main/java/co/cask/cdap/api/templates/plugins/PluginVersion.java]: Oh, this constructor was used in other places in the plugin system, which is not included in this PR.
Sure, I'll add some test on it in this PR.

**Comment by gokulavasan** [comment]
LGTM :+1:

**Comment by chtyim** [comment]
Squashed commit into one to prepare for merge.

**Comment by chtyim** [comment]
Since this PR is only adding a new class and a new test class which are not used anywhere in existing code, there is no way to break existing code. All tests passed in the new test class introduced. Merging this PR.

**Pull request merged**